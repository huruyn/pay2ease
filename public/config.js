window.API_URL = "https://zippy-retail-api-dev.azurewebsites.net";