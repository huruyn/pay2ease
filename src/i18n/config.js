import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

i18n.use(initReactI18next).init({
  fallbackLng: 'en-US',
  lng: 'en-US',
  resources: {
    'en-CA': {
      translations: require('./locales/en-CA/translations.json')
    },
    'en-US': {
      translations: require('./locales/en-US/translations.json')
    },
    'es-US': {
      translations: require('./locales/es-US/translations.json')
    },
    'fr-CA': {
      translations: require('./locales/fr-CA/translations.json')
    },
  },
  ns: ['translations'],
  defaultNS: 'translations'
});

i18n.languages = ['en-CA', 'en-US', 'es-US', 'fr-CA'];

export default i18n;