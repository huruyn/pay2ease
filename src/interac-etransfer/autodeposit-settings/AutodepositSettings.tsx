import { useEffect, useState } from 'react';
import {
  Button,
  Col,
  Form,
  Row,
} from 'react-bootstrap';
import { useMsal } from '@azure/msal-react';
import Api, { User, Account } from '../../api';
import { useAppSelector } from '../../app/hooks';
import { selectUser } from '../../features/user/userSlice';
import CommonPageContainer from '../../common/CommonPageContainer';

const Divider = () => <div className="border-top my-3" />;

const LeftCol = ({ user, accountList }: { user: User, accountList: Account[] }) => (
  <Col xs={9}>
    <Row>
      <Col>
        <h4 className="mt-4">Register for Autodeposit</h4>
      </Col>
    </Row>
    <Row className="mt-4">
      <Col>
        Want the funds you receive to be automatically deposited into your account, without signing
        on or answering security questions? Try Autodeposit. A notification will be sent to your
        contact method whenever a transfer is deposited to the bank account you select. You can
        register up to 5 email addresses and mobile phone numbers for Autodeposit.
      </Col>
    </Row>
    <Row className="align-items-center mt-4">
      <Col xs={3}>Choose contact method:</Col>
      <Col xs={6}>
        <Form.Check type="radio" name="display" id="contact-email" label="Email" checked disabled />
        <Form.Check type="radio" name="display" id="contact-phone" label="Mobile phone number" disabled />
      </Col>
    </Row>
    <Divider />
    <Row className="align-items-center">
      <Col xs={3}>Enter your email Address:</Col>
      <Col xs={6}>
        <Form.Control type="text" value={user.email} />
      </Col>
    </Row>
    <Divider />
    <Row className="align-items-center">
      <Col xs={3}>Automatically Deposit To:</Col>
      <Col xs={6}>
        <Form.Select>
          <option>Select account</option>
          {accountList?.map((account: Account) => (
            <option key={account.id} value={account.id}>
              {account.name}
            </option>
          ))}
        </Form.Select>
      </Col>
    </Row>
    <Divider />
    <Row className="align-items-center">
      <Col xs={3}>Your Legal Name:</Col>
      <Col xs={9}>
        {user.firstName}
        {' '}
        {user.lastName}
      </Col>
    </Row>
    <Row className="mt-4">
      <Col>
        By registering, I understand that my legal name (where applicable) and email address or
        mobile phone number will display to anyone who sends me money through
        {' '}
        <i>Interac</i>
        {' '}
        e-Transfer.
        <br />
        <br />
        You may still be asked to answer a security question when receiving money from financial
        institutions that don&apos;t fully support Autodeposit.
      </Col>
    </Row>
    <Divider />
    <Row>
      <Col className="text-end">
        <Button>Register</Button>
      </Col>
    </Row>
    <Divider />
    <Row>
      <Col>
        <b>Note:</b>
        <br />
        Your use of
        {' '}
        <i>Interac</i>
        {' '}
        e-Transfer is subject to the
        {' '}
        <i>Interac</i>
        {' '}
        e-Transfer Terms and conditions (PDF, 197 KB).
      </Col>
    </Row>
  </Col>
);

const RightCol = () => (
  <Col xs={3}>
    <div className="border p-2 rounded">
      <b>You can also:</b>
      <br />
      Change your contact information &gt;
      <br />
      Change your statement preferences &gt;
      <br />
      View upcoming bill payments and transfers &gt;
    </div>
  </Col>
);

export const AutodepositSettingsPure = ({ user }: { user: User }) => {
  const { instance, accounts } = useMsal();
  const [accountList, setAccountList] = useState<Account[]>([]);

  useEffect(() => {
    new Api(instance, accounts[0])
      .listAccounts()
      .then((res) => {
        setAccountList(res);
      });
  }, [accounts]);
  return (
    <div className="page-content">
      <CommonPageContainer title="Setup AutoDeposit">
        <Row>
          <LeftCol user={user} accountList={accountList} />
          <RightCol />
        </Row>
      </CommonPageContainer>
    </div>
  );
};

export default function AutodepositSettings() {
  const { user } = useAppSelector(selectUser);
  return (user && <AutodepositSettingsPure user={user} />) || null;
}
