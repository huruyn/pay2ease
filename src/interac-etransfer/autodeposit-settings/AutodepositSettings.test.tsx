import '../../i18n/config';
import { Provider } from 'react-redux';
import AutodepositSettings from './AutodepositSettings';
import { store } from '../../app/store';

const ReactTestRenderer = require('react-test-renderer');

describe('AutodepositSettings Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(
      <Provider store={store}>
        <AutodepositSettings />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
