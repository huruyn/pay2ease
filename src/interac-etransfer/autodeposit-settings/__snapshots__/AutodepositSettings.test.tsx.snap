// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutodepositSettings Component matches the snapshot 1`] = `null`;
